src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -1319,11 +1421,13 @@ func (s *commitLogSource) ReadIndex(
 	)
 
 	// Start by reading any available snapshot files.
+	doneReadingSnapshots := s.metrics.index.emitReadingSnapshots()
```

Comment:
If we could do this with a defer somehow that would also be nicer, is there some state it can just read for the whole execution of the function and emit one or zero instead of needing to explicitly emit based on control flow?

Reply:
Yeah there isn't any obvious state we could read to make a decision (any state we could read would have to be added based on control flow.)

I think we either leave it as is, or we can simplify it by not distinguishing between reading commitlogs and snapshotting and just having a gauge for "commitlog bootstrapping data" and "commitlog bootstrapping index"